### PR TITLE
SIT-2382 Migration to MULE 4 - API Led Connectivity System API for Da…

### PR DESCRIPTION
**Pull request by michalspakhotovo**
…tabase

### CODE REVIEW

File: src/main/resources/mule.dev.properties [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# Database Connector configuration
+db.host=
+db.port=
+db.user=
+db.password=
+db.name=
+
+# Properties to be used in Location header
+api.domain=localhost
```
Comment (viktoroooh) [inline src/main/resources/mule.dev.properties]: clear props

File: src/main/resources/mule.prod.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+# Database Connector configuration
+db.host=
+db.port=
+db.user=
+db.password=
+db.name=
+
+# Properties to be used in Location header
+api.domain=localhost
```
Comment (viktoroooh) [inline src/main/resources/mule.prod.properties]: clear props

File: src/main/resources/mule.qa.properties [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+# Database Connector configuration
+db.host=
+db.port=
+db.user=
+db.password=
+db.name=
+
+# Properties to be used in Location header
+api.domain=localhost
```
Comment (viktoroooh) [inline src/main/resources/mule.qa.properties]: clear props

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            <artifactId>mule-db-connector</artifactId>
+            <version>${mule.connectors.version}</version>
+            <classifier>mule-plugin</classifier>
+        </dependency>
+    	<dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>${mysql.version}</version>
+        </dependency>	
+    	<dependency>
```
Comment (viktoroooh) [inline pom.xml]: unnecessary dependency

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-core</artifactId>
+            <version>4.0.4.RELEASE</version>
+        </dependency>
+    	<dependency>
+            <groupId>org.springframework</groupId>
```
Comment (viktoroooh) [inline pom.xml]: unnecessary dependency

File: src/main/resources/mule.dev.properties (hunk 1 above)
Comment (michalspakhotovo) [reply src/main/resources/mule.dev.properties]: values set intentionally as a hint for users as the property could not be clear enough for them

File: src/main/resources/mule.prod.properties (hunk 2 above)
Comment (michalspakhotovo) [reply src/main/resources/mule.prod.properties]: values set intentionally as a hint for users as the property could not be clear enough for them

File: src/main/resources/mule.qa.properties (hunk 3 above)
Comment (michalspakhotovo) [reply src/main/resources/mule.qa.properties]: values set intentionally as a hint for users as the property could not be clear enough for them

File: pom.xml (hunk 4 above)
Comment (michalspakhotovo) [reply pom.xml]: The property has been added by Anypoint Studio during implementation.

File: pom.xml (hunk 5 above)
Comment (michalspakhotovo) [reply pom.xml]: The property has been added by Anypoint Studio during implementation.

File: src/main/resources/mule.prod.properties (hunk 2 above)
Comment (viktoroooh) [reply src/main/resources/mule.prod.properties]: You can provide hints in comments (the same approach we are using in 3.8 branch of apiled system APIs for DB and SFDC): 

# example: api.domain=system-api-db.cloudhub.io
api.domain=
# example: api.basePath=/api
api.basePath=

but its about personal preference (IMHO: clear props)

File: src/main/mule/system-api.xml [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+http://www.mulesoft.org/schema/mule/db http://www.mulesoft.org/schema/mule/db/current/mule-db.xsd
+http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd">  
+    <http:listener-config name="system-api-httpListenerConfig">
+        <http:listener-connection host="0.0.0.0" port="8081" />
```
Comment (viktoroooh) [inline src/main/mule/system-api.xml]: property placeholder for port

File: src/main/mule/system-api.xml [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+		</apikit:flow-mappings>
+	</apikit:config>
+	<db:config name="Database_Config" doc:name="Database Config" doc:id="f75ca3de-9f51-4f8e-8518-14aea2af9c78" >
+		<db:my-sql-connection host="${db.host}" port="${db.port}" user="${db.user}" password="${db.password}" database="${db.name}" />
+	</db:config>
+	<configuration-properties file="mule-artifact.properties" doc:name="Configuration properties" doc:id="9ceefa30-f66a-478f-94e3-561022a09610" />
+	<configuration-properties file="common.properties" doc:name="Configuration properties" doc:id="fb6ccc25-3b1e-4ce3-9129-39d7cd3995d2" />
+	<configuration-properties file="mule.${mule.env}.properties" doc:name="Configuration properties" doc:id="320d97b0-24fa-4733-8a23-7746b800ae6b" />
+	<flow name="system-api-main">
+        <http:listener config-ref="system-api-httpListenerConfig" path="/api/*" doc:name="Listener">
```
Comment (viktoroooh) [inline src/main/mule/system-api.xml]: property placeholder for basePath (path in http listener).

Currently it is used in location header ('Location':'http://${api.domain}:${http.port}/${api.basePath}/accounts/' ++ vars.id)

**Pull request merged**